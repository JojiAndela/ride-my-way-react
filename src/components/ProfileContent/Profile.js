import React, { Component } from 'react';
import { HashRouter, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import classes from './Profile.scss';
import Cards from './Cards/Cards';


class Profile extends Component {

  render() {
    let profile = null;
    if(this.props.user){
    const { currentUser, ridesGiven, ridesTaken } = this.props.user;
    return (
      <div>
      <div className={classes.Content}>
        <div className={classes.User}>
        <h1>{currentUser.username}</h1>
      <h4>{currentUser.email}</h4>
        </div>
        <div className={classes.Details}>
          <Link to="#rides">Total rides Given:  <span>{ridesGiven.length}</span></Link>
          <Link to="#rides">Total rides Taken:  <span>{ridesTaken.length}</span></Link>
          <Link to="#s">Total rides Request:  <span>0</span></Link>
          <Link to="#a">Total rides Request by you:  <span>0</span></Link>
        </div>
      </div>
        <div className={classes.Rides} id="#rides">
        <Cards title="Rides Given" rides={ridesGiven} />
      <Cards title="Rides Taken" rides={ridesTaken}/>
        </div>
      </div>
    );
  }
    return (
      <HashRouter>
      {profile}
      </HashRouter>
    );
  }
}

Profile.propTypes ={
  user: PropTypes.object.isRequired,
  // ridesReqSent: PropTypes.number.isRequired,
  // ridesReqReceived: PropTypes.number.isRequired,

}

const mapStateToProps = state => ({
  user: state.user.user
});


export default connect(mapStateToProps)(Profile);
