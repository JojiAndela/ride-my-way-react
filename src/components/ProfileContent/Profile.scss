.Content{
  width: 90%;
  margin: 20px auto;
  border: 1px solid #ccc;
  border-radius: 5px;
  box-shadow: 2px 5px 8px  #ccc;
  padding: 20px;
  display: flex;

  &:hover{
    margin-top: 25px;
    box-shadow: 2px 2px 4px  #ccc;
  }
  .User{
    margin: 10px;
    flex: 1;
    text-align: center;
    color: #333;
  }
  .Details{
    border-left: 1px solid #ccc;
    padding: 20px 40px;
    flex: 2;
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    a{
      text-decoration: none;
      color: #333;
      // width: 50%;

      &:hover{
        color: teal;
        border-bottom: 1px #fff solid;
      }

      span{
        float: right;
      }
    }

  }
}

.Rides{
  display: inline-grid;
  grid-template-columns: auto auto;
  grid-gap: 20px;
  width: 95%;
  margin: 20px auto;
  padding: 20px;

}
