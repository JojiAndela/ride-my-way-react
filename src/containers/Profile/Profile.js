import React, { Component } from 'react';
import ProfileContent from 'Components/ProfileContent/Profile';

class Profile extends Component {
  render() {
    return (
      <div>
        <ProfileContent
        ridesTaken={2}
        ridesGiven={3}
        ridesReqReceived={2}
        ridesReqSent={0}
        />
      </div>
    );
  }
}

export default Profile;
