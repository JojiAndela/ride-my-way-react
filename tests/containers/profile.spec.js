import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

// Component to be tested
import Profile from '../../src/containers/Profile/Profile';

const initialState ={
  user: {
    loading: false,
    user: {
      currentUser: "user",
      ridesGiven: [],
      ridesTaken: []
    },
  }
}


const middleware = [thunk];
const mockStore = configureStore(middleware);
let wrapper;
let store;
const getProfileAction = jest.fn();
describe('Profile', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Profile
            getProfileAction={getProfileAction}
          />
        </BrowserRouter>
      </Provider>
    );
  });
  it('should render the profile page correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
