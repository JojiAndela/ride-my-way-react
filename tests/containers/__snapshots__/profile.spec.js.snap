// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render the profile page correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Profile)
        getProfileAction={[MockFunction]}
      >
        <Profile
          getProfileAction={[Function]}
          loading={false}
        >
          <div
            className="profile"
          >
            <Connect(Profile)>
              <Profile
                dispatch={[Function]}
                user={
                  Object {
                    "currentUser": "user",
                    "ridesGiven": Array [],
                    "ridesTaken": Array [],
                  }
                }
              >
                <div>
                  <div
                    className="Content"
                  >
                    <div
                      className="User"
                    >
                      <h1 />
                      <h4 />
                    </div>
                    <div
                      className="Details"
                    >
                      <Link
                        replace={false}
                        to="#rides"
                      >
                        <a
                          href="/#rides"
                          onClick={[Function]}
                        >
                          Total rides Given:  
                          <span>
                            0
                          </span>
                        </a>
                      </Link>
                      <Link
                        replace={false}
                        to="#rides"
                      >
                        <a
                          href="/#rides"
                          onClick={[Function]}
                        >
                          Total rides Taken:  
                          <span>
                            0
                          </span>
                        </a>
                      </Link>
                      <Link
                        replace={false}
                        to="#s"
                      >
                        <a
                          href="/#s"
                          onClick={[Function]}
                        >
                          Total rides Request:  
                          <span>
                            0
                          </span>
                        </a>
                      </Link>
                      <Link
                        replace={false}
                        to="#a"
                      >
                        <a
                          href="/#a"
                          onClick={[Function]}
                        >
                          Total rides Request by you:  
                          <span>
                            0
                          </span>
                        </a>
                      </Link>
                    </div>
                  </div>
                  <div
                    className="Rides"
                    id="#rides"
                  >
                    <withRouter(cards)
                      rides={Array []}
                      title="Rides Given"
                    >
                      <Route>
                        <cards
                          history={
                            Object {
                              "action": "POP",
                              "block": [Function],
                              "createHref": [Function],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "length": 1,
                              "listen": [Function],
                              "location": Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              },
                              "push": [Function],
                              "replace": [Function],
                            }
                          }
                          location={
                            Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            }
                          }
                          match={
                            Object {
                              "isExact": true,
                              "params": Object {},
                              "path": "/",
                              "url": "/",
                            }
                          }
                          rides={Array []}
                          title="Rides Given"
                        >
                          <div
                            className="cards"
                          >
                            <h3>
                               
                              Rides Given
                               
                            </h3>
                            <div
                              className="content"
                            >
                              <smallCard
                                clicked={[Function]}
                                status="You have got no ride here"
                                title=" No Ride Yet"
                              >
                                <div
                                  className="card"
                                  onClick={[Function]}
                                >
                                  <h2>
                                     No Ride Yet
                                  </h2>
                                  <p>
                                    You have got no ride here
                                  </p>
                                </div>
                              </smallCard>
                            </div>
                          </div>
                        </cards>
                      </Route>
                    </withRouter(cards)>
                    <withRouter(cards)
                      rides={Array []}
                      title="Rides Taken"
                    >
                      <Route>
                        <cards
                          history={
                            Object {
                              "action": "POP",
                              "block": [Function],
                              "createHref": [Function],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "length": 1,
                              "listen": [Function],
                              "location": Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              },
                              "push": [Function],
                              "replace": [Function],
                            }
                          }
                          location={
                            Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            }
                          }
                          match={
                            Object {
                              "isExact": true,
                              "params": Object {},
                              "path": "/",
                              "url": "/",
                            }
                          }
                          rides={Array []}
                          title="Rides Taken"
                        >
                          <div
                            className="cards"
                          >
                            <h3>
                               
                              Rides Taken
                               
                            </h3>
                            <div
                              className="content"
                            >
                              <smallCard
                                clicked={[Function]}
                                status="You have got no ride here"
                                title=" No Ride Yet"
                              >
                                <div
                                  className="card"
                                  onClick={[Function]}
                                >
                                  <h2>
                                     No Ride Yet
                                  </h2>
                                  <p>
                                    You have got no ride here
                                  </p>
                                </div>
                              </smallCard>
                            </div>
                          </div>
                        </cards>
                      </Route>
                    </withRouter(cards)>
                  </div>
                </div>
              </Profile>
            </Connect(Profile)>
          </div>
        </Profile>
      </Connect(Profile)>
    </Router>
  </BrowserRouter>
</Provider>
`;
